### ISSUE TITLE
"Potential incomplete link" warnings generated for checklists

### ISSUE THREAD

**Issue by brendanzab** [Michael-F-Bryan/mdbook-linkcheck#45]
I get the following warning when I run mdbook-linkcheck on a markdown file with a checklist:

```
warning: Potential incomplete link
   ┌─ development/roadmap.md:19:5
   │
19 │   - [x] Comments
   │     ^^^ Did you forget to define a URL for `x`?
   │
   = hint: declare the link's URL. For example: `[x]: http://example.com/`
```

It seems to be fine with unchecked list items, eg: `- [ ] Blah`.

**Comment by ghost-in-the-zsh** [comment]
I'm getting this warning when writing MathJax equations, etc. For example:

```
2022-09-11 22:51:43 [INFO] (mdbook::renderer): Invoking the "linkcheck" renderer
warning: Potential incomplete link
┌─ ch00/04-special-products.md:37:3
│  
37 │   \\[
│ ╭───^
38 │ │     x = \frac{
39 │ │             -b \pm \sqrt{ (b^2 - 4ac) }
40 │ │         }{
41 │ │             2a
42 │ │         }
43 │ │ \\]
│ ╰───^ Did you forget to define a URL for ` x = \frac{ -b \pm \sqrt{ (b^2 - 4ac) } }{ 2a } \\`?
│  
= hint: declare the link's URL. For example: `[ x = \frac{ -b \pm \sqrt{ (b^2 - 4ac) } }{ 2a } \\]: http://example.com/`
```

I recently got back to working on personal references after a rather long break (since 2019), and these warnings weren't showing up back then, but they show up now. Is this expected or not? Any suggestions or workarounds?
